src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,56 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+/**
+ * Represents a Person's contact in the address book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidContact(String, String)}
+ */
+public class Contact {
+
+    public final String value;
+    protected boolean isPrivate;
+
+    /**
+     * Validates given contact.
+     *
+     * @throws IllegalValueException if given contact string is invalid.
+     */
+    public Contact(String contact, boolean isPrivate, String regex, String constraints) throws IllegalValueException {
+        this.isPrivate = isPrivate;
+        String trimmedContact = contact.trim();
+        if(!isValidContact(trimmedContact, regex)){
+            throw new IllegalValueException(constraints);
+        }
+        this.value = trimmedContact;
+    }
+
+    /**
+     * Returns true if the given string is a valid person contact.
+     */
+    public static boolean isValidContact(String test, String regex) {
+        return test.matches(regex);
+    }
+
+    @Override
+    public String toString() {
+        return value;
+    }
+
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || ((other.getClass() == this.getClass()) // instanceof handles nulls
+                && this.value.equals(((Contact) other).value)); // state check
```

Comment:
Ouch! you see that you are comparing parent class with one other child class instance? 
Think how you would alleviate this problem.